src/test/fio/README.md

```diff
@@ -86,3 +86,31 @@ If you installed ceph in any other place (cmake -DCMAKE_INSTALL_PREFIX=${CEPH_IN
     rados=yes make
 
 "-ltcmalloc" is necessary if ceph was compiled with tcmalloc.
+
+Messenger
+---------
+
+This fio engine allows you to test CEPH messenger transport layer, without
+any disk activities involved.
+
+To build fio_ceph_messenger:
+```
+  mkdir build && cd build
+  cmake -DWITH_FIO=ON -DFIO_INCLUDE_DIR=/path/to/fio -DCMAKE_BUILD_TYPE=Release /path/to/ceph
+  make -C src/test/fio install
+```
+If you install the ceph libraries to a location that isn't in your
+LD_LIBRARY_PATH, be sure to add it:
+
+    export LD_LIBRARY_PATH=/path/to/install/lib
+
+To view the fio options specific to the messenger engine:
+
+    ./fio --enghelp=libfio_ceph_messenger.so
+
+The ceph_conf_file= option requires a ceph configuration file (ceph.conf),
+see ceph-messenger.conf and ceph-messenger.fio for details.
+
+To run:
+
+    ./fio ./ceph-messenger.fio
```

Comment:
Take a look into ceph-messenger.fio config, there are two sections client and server. Comment the client section out on server and the server section on client, specify correct hostname on the client and you are done.

Reply:
[global]
size=4k
ioengine=libfio_ceph_messenger.so
ceph_conf_file=ceph-messenger.conf
ms_type=async+posix # or async+dpdk or async+rdma
[server]
hostname=10.0.0.7
port=5555
receiver=1

If not set size, fio print "server: you need to specify size=". But what's mean of size for server? And i change this 1G.  when it existed , the client did existed.  How to make server don't return before client existed?

Reply:
Since the whole server thingy is a hack for a fio, the main requirement is the following: global configuration options should be equal for both client and server sides, thus it is not intended that server accepts many clients during the whole run cycle. So let's put it like that: this is not a "server" in general meaning, this "server" accepts only one client, receives exactly amount of bytes specified in size= option and then exits.

Of course server side can be a standalone application which accepts clients, handshakes with them (accepts configration) and never exits, but I simply do not want another server tool for only one purpose: accepts fio connections. As a next step that can be easily done, but for now another fio process acts as a server.

Reply:
how about perf_msgr_server.cc in https://github.com/ceph/ceph/tree/master/src/test/msgr?

Reply:
It does couple of things which I would like to avoid, change or do in more efficient way: 
1. I do not need any threads, that increases latency. Request should be replied back immediately, without offloading it to another thread.
2. Server side also should keep a queue, i.e. request can't be processed immediately if no free slot is available, that is natural in IO stack to propagate the backpressure. 
3. You might already noticed, that client side embeds just a pointer inside a request, so to avoid any search algorithms to map reply->request I expect server to send me this pointer back, for that reason server side has to do "req->finish_decode()", which is not true for perf_msgr_server.